### PR TITLE
HyperV IP injection support

### PR DESCRIPTION
**Pull request by johnsongeorge-w**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] All package sources are available
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Ready to merge

---

###### Summary 

Support IP injection.
The current hv_set_ifconfig script from kernel source is used in Mariner. That is a sample script and based on RHEL, this script should be updated or customized based on the network manager used by distribution. Created a new hv_set_ifconfig script to configure the network settings for Mariner.

###### Change Log  
Added a new script hv_set_ifconfig
``` Package: hypervkvpd```

###### Does this affect the toolchain?  
**NO**

###### Test Methodology
```
1. Tested locally by copying the script in Deployed VM
2. Tested using the newly created hypervkvpd package. 
    a. Build a new package(hypervkvpd)
    b. Deploy a VM in HyperV Manager
    c. Boot the VM & Deploy package
    d. Run **NET-IP-INJECTION** from LISA test
    e. Test completed successfully.
```

### CODE REVIEW

**Comment by nicogbg** [comment]
could you check why lint check failed ? You can use https://dev.azure.com/mariner-org/mariner/_git/spec-cleaner-fork

File: SPECS/hyperv-daemons/hyperv-daemons.spec [hunk 1]
```diff
@@ -19,6 +19,7 @@ Source0:        https://github.com/microsoft/WSL2-Linux-Kernel/archive/linux-msf
 # HYPERV KVP DAEMON
 Source1:        hypervkvpd.service
 Source2:        hypervkvp.rules
+Source3:        hv_set_ifconfig
```
Comment (nicogbg) [inline SPECS/hyperv-daemons/hyperv-daemons.spec:22]: Is that expected that there is no .sh extension?

Comment (nicogbg) [inline SPECS/hyperv-daemons/hyperv-daemons.spec:22]: you should bump up the Release number above -> Release:        3%{?dist}
and add a change log below

**Review by nicogbg — CHANGES_REQUESTED** [review]

File: SPECS/hyperv-daemons/hyperv-daemons.spec (hunk 1 above)
Comment (johnsongeorge-w) [reply SPECS/hyperv-daemons/hyperv-daemons.spec:22]: Yes. The actual target file is hv_set_ifconfig

Comment (johnsongeorge-w) [reply SPECS/hyperv-daemons/hyperv-daemons.spec:22]: Fixed. Bumped up the version and updated the change logs. Thanks!

**Comment by johnsongeorge-w** [comment]
> could you check why lint check failed ? You can use https://dev.azure.com/mariner-org/mariner/_git/spec-cleaner-fork

Thanks. The lint check is now fixed and it passed in PR checker.

**Review by nicogbg — APPROVED** [review]

File: SPECS/hyperv-daemons/hyperv-daemons.spec [hunk 2]
```diff
@@ -140,7 +141,7 @@ install -p -m 0644 %{SOURCE202} %{buildroot}%{_udevrulesdir}/%{udev_prefix}-hype
 mkdir -p %{buildroot}%{_libexecdir}/%{hv_kvp_daemon}
 install -p -m 0755 tools/hv/hv_get_dhcp_info.sh %{buildroot}%{_libexecdir}/%{hv_kvp_daemon}/hv_get_dhcp_info
 install -p -m 0755 tools/hv/hv_get_dns_info.sh %{buildroot}%{_libexecdir}/%{hv_kvp_daemon}/hv_get_dns_info
-install -p -m 0755 tools/hv/hv_set_ifconfig.sh %{buildroot}%{_libexecdir}/%{hv_kvp_daemon}/hv_set_ifconfig
```
Comment (PawelWMS) [inline SPECS/hyperv-daemons/hyperv-daemons.spec:143]: Wanted to make sure here: how much does the new script differ from the default one? I'm mostly concerned about breaking any existing functionality, so I'd like to better understand what we're changing.

**Review by PawelWMS — APPROVED** [review]

File: SPECS/hyperv-daemons/hyperv-daemons.spec (hunk 2 above)
Comment (johnsongeorge-w) [reply SPECS/hyperv-daemons/hyperv-daemons.spec:143]: Its totally different. That is the core reason for this PR. We cannot use the existing hv_set_ifconfig.sh and it needs to be adapted as per the network manager supported by the distro.

Comment (PawelWMS) [reply SPECS/hyperv-daemons/hyperv-daemons.spec:143]: I see, thanks for the explanation!

File: SPECS/hyperv-daemons/hyperv-daemons.spec [hunk 3]
```diff
@@ -107,16 +108,16 @@ Contains tools and scripts useful for Hyper-V guests.
 pushd tools/hv
 
 # HYPERV KVP DAEMON
-%{__cc} %{optflags} -c hv_kvp_daemon.c
-%{__cc} $RPM_LD_FLAGS  hv_kvp_daemon.o -o %{hv_kvp_daemon}
+gcc %{optflags} -c hv_kvp_daemon.c
```
[the remark this answers is not in the document]
Comment (johnsongeorge-w) [reply SPECS/hyperv-daemons/hyperv-daemons.spec]: @christopherco Revert the spec lint fix.

**Review by christopherco — APPROVED** [review]

**Pull request merged**